test/jdk/java/nio/channels/DatagramChannel/Connect.java

```diff
@@ -135,31 +131,31 @@ int port() {
             return dc.socket().getLocalPort();
         }
 
-        public void throwException() throws Exception {
-            if (e != null)
-                throw e;
-        }
-
         public void run() {
             try {
                 // Listen for a message
                 ByteBuffer bb = ByteBuffer.allocateDirect(100);
+                log.println("Reactor waiting to receive");
                 SocketAddress sa = dc.receive(bb);
                 bb.flip();
-                CharBuffer cb = Charset.forName("US-ASCII").
-                newDecoder().decode(bb);
-                log.println("From Actor: "+sa+ " said " +cb);
+                CharBuffer cb = StandardCharsets.US_ASCII.
+                        newDecoder().decode(bb);
+                log.println("Reactor received from Actor at" + sa +  ": " + cb);
 
                 // Reply to sender
                 dc.connect(sa);
                 bb.flip();
+                log.println("Reactor attempting to write: " + dc.getRemoteAddress().toString());
                 dc.write(bb);
 
                 // Clean up
                 dc.disconnect();
                 dc.close();
```

Comment:
Can you move the dc.close to the finally block so the socket is closed when there is an exception thrown.